Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java

```diff
@@ -0,0 +1,428 @@
+
+    static final String LAST_USED_PAYMENT_METHOD_TYPE =
+            "com.braintreepayments.api.dropin.LAST_USED_PAYMENT_METHOD_TYPE";
+
+    private final BraintreeClient braintreeClient;
+    private final PaymentMethodClient paymentMethodClient;
+    private final GooglePayClient googlePayClient;
+    private final PayPalClient payPalClient;
+    private final VenmoClient venmoClient;
+    private final CardClient cardClient;
+    private final UnionPayClient unionPayClient;
+
+    private final DropInRequest dropInRequest;
+    private final ThreeDSecureClient threeDSecureClient;
+    private final DataCollector dataCollector;
+
+    private static DropInClientParams createDefaultParams(Context context, String authorization, DropInRequest dropInRequest) {
+        BraintreeClient braintreeClient = new BraintreeClient(context, authorization);
+        return new DropInClientParams()
+                .dropInRequest(dropInRequest)
+                .braintreeClient(braintreeClient)
+                .threeDSecureClient(new ThreeDSecureClient(braintreeClient))
+                .paymentMethodClient(new PaymentMethodClient(braintreeClient))
+                .payPalClient(new PayPalClient(braintreeClient))
+                .venmoClient(new VenmoClient(braintreeClient))
+                .cardClient(new CardClient(braintreeClient))
+                .unionPayClient(new UnionPayClient(braintreeClient))
+                .dataCollector(new DataCollector(braintreeClient))
+                .googlePayClient(new GooglePayClient(braintreeClient));
+    }
+
+    public DropInClient(Context context, String authorization, DropInRequest dropInRequest) {
+        this(context, authorization, null, dropInRequest);
+    }
+
+    DropInClient(Context context, String authorization, String sessionId, DropInRequest dropInRequest) {
+        // TODO: instantiate a BraintreeClient with the input sessionId
+        this(createDefaultParams(context, authorization, dropInRequest));
+    }
+
+    @VisibleForTesting
+    DropInClient(DropInClientParams params) {
+        this.dropInRequest = params.getDropInRequest();
+        this.braintreeClient = params.getBraintreeClient();
+        this.googlePayClient = params.getGooglePayClient();
+        this.paymentMethodClient = params.getPaymentMethodClient();
+        this.threeDSecureClient = params.getThreeDSecureClient();
+        this.payPalClient = params.getPayPalClient();
+        this.venmoClient = params.getVenmoClient();
+        this.cardClient = params.getCardClient();
+        this.unionPayClient = params.getUnionPayClient();
+        this.dataCollector = params.getDataCollector();
+    }
+
+    Authorization getAuthorization() {
+        return braintreeClient.getAuthorization();
+    }
+
+    void getConfiguration(ConfigurationCallback callback) {
+        braintreeClient.getConfiguration(callback);
+    }
+
+    void sendAnalyticsEvent(String eventFragment) {
```

Comment:
Should this be non-public / hidden from merchants?

Reply:
This is package-privates so we should be good. It's used outside of this class so can't make it fully private.

Reply:
Ah, the default is package-private. Thank you!